app/src/full/java/io/homeassistant/companion/android/settings/views/SettingsWearMainView.kt

```diff
@@ -0,0 +1,77 @@
+package io.homeassistant.companion.android.settings.views
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.util.Log
+import androidx.activity.compose.setContent
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
+import com.google.android.gms.wearable.DataClient
+import com.google.android.gms.wearable.DataEvent
+import com.google.android.gms.wearable.DataEventBuffer
+import com.google.android.gms.wearable.DataMapItem
+import com.google.android.gms.wearable.Node
+import com.google.android.gms.wearable.Wearable
+import dagger.hilt.android.AndroidEntryPoint
+import io.homeassistant.companion.android.common.data.integration.IntegrationRepository
+import io.homeassistant.companion.android.settings.SettingsWearViewModel
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class SettingsWearMainView : AppCompatActivity(), DataClient.OnDataChangedListener {
+
+    private val settingsWearViewModel by viewModels<SettingsWearViewModel>()
+
+    @Inject
+    lateinit var integrationUseCase: IntegrationRepository
+
+    companion object {
+        private const val TAG = "SettingsWearDevice"
+        private var currentNodes = setOf<Node>()
+        const val LANDING = "Landing"
+        const val FAVORITES = "Favorites"
+
+        fun newInstance(context: Context, wearNodes: Set<Node>): Intent {
+            currentNodes = wearNodes
+            return Intent(context, SettingsWearMainView::class.java)
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        val activity = this
+        setContent {
+            LoadSettingsHomeView(settingsWearViewModel, currentNodes.first().displayName, activity)
+        }
+        settingsWearViewModel.init(integrationUseCase)
+    }
+
+    override fun onResume() {
+        super.onResume()
+        Wearable.getDataClient(this).addListener(this)
+        Thread { settingsWearViewModel.findExistingFavorites(this) }.start()
+        Thread { settingsWearViewModel.requestFavorites(this) }.start()
+    }
+
+    override fun onPause() {
+        super.onPause()
+        Wearable.getDataClient(this).removeListener(this)
+    }
+
+    override fun onDataChanged(dataEvents: DataEventBuffer) {

```

Comment:
This belongs in the View Model I think.

Reply:
I moved over the logic but we still need `onPause` and `onResume` from the activity to properly set the listener

Reply:
moved all logic to the view model and just passing in the the proper calls in `onResume` and `onPause` now